maintenance/code-backports.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+# silence debug output from VCS
+beQuiet="--quiet"
+#beQuiet=""
+version=`basename "$1" | sed s/squid-//`
+vnext=$(( $version + 1 ))
+srcbranch=v$vnext
+
+cd $1
+
+# Dev branch does not receive backports
+test -f .BASE && exit 0
+
+# clean the current workspace
+gitCleanWorkspace ()
+{
+  git clean $beQuiet -xdf --exclude="\.BASE"
+  git checkout $beQuiet -- .
+}
+
+if ! test -d .git ; then
+  echo "ERROR: missing git repository"
+  exit 1;
+fi
+if ! test -d ../squid-$vnext ; then
+  echo "ERROR: missing git repository to backport from"
+  exit 1;
+fi
+if test -e ../squid-$vnext/.BASE ; then
+  srcbranch=`cat ../squid-$vnext/.BASE`
+fi
```

Comment:
Instead of this convoluted `.BASE` hack and related manipulation, I suggest making `srcbranch` a script parameter with `master` (or the main official repository branch name if you want to compute that to avoid hard-coding `master`) as its default value.

Reply:
Where do you propose that script parameter comes from? The script running this script only has the "/path/to/repo" and "vN" branch name available. The requirement to use "master" for development prohibits using `git branch` to determine if there is a newer vN branch existing.